### ISSUE TITLE
Can't delete Route53 CNAME record

### ISSUE THREAD

**Issue by smiller171** [ansible/ansible-modules-core#551]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.8.2
  configured module search path = None
##### Environment:

Running from Ubuntu 14.04 Ansible Tower against AWS API
##### Summary:

When attempting to delete a Route53 CNAME record, (get and then delete) using the route53 module, it always fails saying the values provided do not match.
##### Steps To Reproduce:

```

---
- name: get record
  route53:
    command: get
    zone: foo.com
    record: "{{stack}}.foo.com"
    type: CNAME
  register: rec

- name: debug rec var
  debug: var=rec

- name: Delete URL
  route53:
    command: delete
    zone: foo.com
    record: "{{ rec.set.record }}"
    type: "{{ rec.set.type }}"
    value: "{{ rec.set.value }}"
```
##### Expected Results:

```
TASK: [delete_workbench | get record] ***************************************** 
ok: [localhost]

TASK: [delete_workbench | debug rec var] ************************************** 
ok: [localhost] => {
    "rec": {
        "changed": false, 
        "invocation": {
            "module_args": "", 
            "module_name": "route53"
        }, 
        "set": {
            "record": "ansible-2.projectcobe.com.", 
            "ttl": "300", 
            "type": "CNAME", 
            "value": "ansible-2-workbench-ELB-752533913.us-east-1.elb.amazonaws.com", 
            "values": [
                "ansible-2-workbench-ELB-752533913.us-east-1.elb.amazonaws.com"
            ], 
            "zone": "projectcobe.com."
        }
    }
}

TASK: [delete_workbench | Delete URL] ***************************************** 
changed: [localhost]

PLAY RECAP ******************************************************************** 

localhost                  : ok=2    changed=1    unreachable=0    failed=0

##### Actual Results:

TASK: [delete_workbench | get record] ***************************************** 
ok: [localhost]

TASK: [delete_workbench | debug rec var] ************************************** 
ok: [localhost] => {
    "rec": {
        "changed": false, 
        "invocation": {
            "module_args": "", 
            "module_name": "route53"
        }, 
        "set": {
            "record": "ansible-2.projectcobe.com.", 
            "ttl": "300", 
            "type": "CNAME", 
            "value": "ansible-2-workbench-ELB-752533913.us-east-1.elb.amazonaws.com", 
            "values": [
                "ansible-2-workbench-ELB-752533913.us-east-1.elb.amazonaws.com"
            ], 
            "zone": "projectcobe.com."
        }
    }
}

TASK: [delete_workbench | Delete URL] ***************************************** 
failed: [localhost] => {"failed": true}
msg: Tried to delete resource record set [name='ansible-2.projectcobe.com.', type='CNAME'] but the values provided do not match the current values

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/awx/delete-environment.yaml.retry

localhost                  : ok=3    changed=0    unreachable=0    failed=1
```

**Comment by alexclifford** [comment]
I believe you will get this behaviour if you have a non-default TTL set for the record. I also came across this problem and was pulling my hair out. I debugged it and found that the problem is at https://github.com/ansible/ansible-modules-core/blob/devel/cloud/amazon/route53.py#L266 where ttl_in is always specified (for both create and delete). The work around is to set the TTL value in the delete that you just obtained from your get:

```
---
- name: get record
  route53:
    command: get
    zone: foo.com
    record: "{{stack}}.foo.com"
    type: CNAME
  register: rec

- name: debug rec var
  debug: var=rec

- name: Delete URL
  route53:
    command: delete
    zone: foo.com
    record: "{{ rec.set.record }}"
    ttl: "{{ rec.set.ttl }}"
    type: "{{ rec.set.type }}"
    value: "{{ rec.set.value }}"
```

I've created a Pull Request (https://github.com/ansible/ansible-modules-core/pull/659) back to ansible-modules-core to see if someone wants to change this behaviour.

**Comment by alexclifford** [comment]
Actually, looking further it looks like because boto sets a default TTL of 600 you have to specify it all the time. My fix probably doesn't help that. The module would have to be altered to look up the TTL for you and set it during its delete if we really wanted to make it optional from the delete command.

**Comment by smiller171** [comment]
Alex,

Doing the TTL lookup as in your example works. I think that all that needs
to be done is to put that into the documented example.

**Comment by alexclifford** [comment]
Yeah I agree, the way boto API works and the route53 module has been created, the TTL is required. I'll throw a PR for the docs update.

**Comment by smiller171** [comment]
@alexclifford @bcoca I think this issue is fixed by #668 and can be closed

**Comment by bcoca** [comment]
closing as per above
